### PR TITLE
select aarch64 specific tests on Apple arm64

### PR DESCRIPTION
**Pull request by bjacob**
I was wondering why we didn't catch on Mac/ARM CI that aarch64 regression from llvm-integrate, which we only saw on Android device tests (#12546). The reason was that our CMake logic only enabled those tests when CMAKE_SYSTEM_PROCESSOR is "aarch64", but on Apple it is "arm64" instead. We have some other places in the code that are already checking for both values, but this place was missing it.

@wangkuiyi I haven't tested this locally as I haven't set up my Mac for local development.

### CODE REVIEW

**Comment by wangkuiyi** [comment]
@bjacob I can checkout this branch and run it on my M1 MacBook Pro. Shoud I run `cmake --build ../iree-build --target iree-run-tests` to verify this pull request works?

File: build_tools/cmake/iree_check_test.cmake [hunk 1]
```diff
@@ -308,7 +308,17 @@ _FILTER_ARCH:_TARGET_CPU_FEATURES. Got: ${_INPUT_TARGET_CPU_FEATURES}")
   # TARGET_CPU_FEATURES_VARIANT is of the form _FILTER_ARCH:_TARGET_CPU_FEATURE.
   list(GET _COMPONENTS 0 _FILTER_ARCH)
   list(GET _COMPONENTS 1 _TARGET_CPU_FEATURES)
-  if(_FILTER_ARCH STREQUAL CMAKE_SYSTEM_PROCESSOR)
+  # Canonicalize the architecture name - some architectures have multiple
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_check_test.cmake:311]: I wonder if we should do some canonicalization at the top level so every place that checks this doesn't have to repeat this logic (that would be a separate PR though).

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/cmake/iree_check_test.cmake (hunk 1 above)
Comment (bjacob) [reply build_tools/cmake/iree_check_test.cmake:311]: Was thinking the same. Definitely error prone and going to haunt us again. I might send something if no one gets there first :-)

**Comment by bjacob** [comment]
> @bjacob I can checkout this branch and run it on my M1 MacBook Pro. Shoud I run `cmake --build ../iree-build --target iree-run-tests` to verify this pull request works?

@wangkuiyi  That would be fantastic, if you have time. And then, from the ctest output, check if you can see any tests named `e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod`: if you run

```
ctest -R e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod -V
```

You should see it actually doing matmul tests, with output looking like this:

```
25: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
25: Matmul shape (MxKxN): 123x456x789
25: --- CALL[module.matmul_123x456xi8_times_456x789xi8_into_123x789xi32] ---
25: Matmul shape (MxKxN): 123x456x789
25: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
25: Matmul shape (MxKxN): 654x321x234
25: --- CALL[module.matmul_654x321xi8_times_321x234xi8_into_654x234xi32] ---
25: Matmul shape (MxKxN): 654x321x234
25: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
25: Matmul shape (MxKxN): 1x1000x1000
25: --- CALL[module.matmul_1x1000xi8_times_1000x1000xi8_into_1x1000xi32] ---
25: Matmul shape (MxKxN): 1x1000x1000
25: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
25: Matmul shape (MxKxN): 1000x1000x1
25: --- CALL[module.matmul_1000x1000xi8_times_1000x1xi8_into_1000x1xi32] ---
25: Matmul shape (MxKxN): 1000x1000x1
```

If it compains that no such tests exists then that would indicate a problem in the CMake logic being touched in this PR.

**Comment by wangkuiyi** [comment]
Hi @bjacob , here is what I got on my M1 MacBook Pro after 
1. checking out your feature branch,
2. building the code following the [CI configuration](https://github.com/openxla/iree/blob/3d6c54e0367d2777bd020d8a689ce62e6337a34e/.github/workflows/ci.yml#L263-L294), and
3. running the command `ctest -R e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod -V` in the build directory:
```
test 34
    Start 34: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod

34: Test command: /Users/y/w/l/iree/build-macos/tools/iree-e2e-matmul-test "/Users/y/w/l/iree/build-macos/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod.yaml" "--device=local-task"
34: Working Directory: /Users/y/w/l/iree/build-macos/tests/e2e/matmul
34: Environment variables:
34:  TEST_TMPDIR=/Users/y/w/l/iree/build-macos/test_tmpdir/iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod_test_tmpdir
34:  IREE_BINARY_DIR=/Users/y/w/l/iree/build-macos
34:  IREE_VULKAN_DISABLE=1
34: Test timeout computed to be: 60
34: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
34: Matmul shape (MxKxN): 123x456x789
34: --- CALL[module.matmul_123x456xi8_times_456x789xi8_into_123x789xi32] ---
34: Matmul shape (MxKxN): 123x456x789
34: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
34: Matmul shape (MxKxN): 654x321x234
34: --- CALL[module.matmul_654x321xi8_times_321x234xi8_into_654x234xi32] ---
34: Matmul shape (MxKxN): 654x321x234
34: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
34: Matmul shape (MxKxN): 1x1000x1000
34: --- CALL[module.matmul_1x1000xi8_times_1000x1000xi8_into_1x1000xi32] ---
34: Matmul shape (MxKxN): 1x1000x1000
34: --- CALL[module.matmul_DYNxDYNxi8_times_DYNxDYNxi8_into_DYNxDYNxi32] ---
34: Matmul shape (MxKxN): 1000x1000x1
34: --- CALL[module.matmul_1000x1000xi8_times_1000x1xi8_into_1000x1xi32] ---
34: Matmul shape (MxKxN): 1000x1000x1
1/1 Test #34: iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod ...   Passed    0.11 sec

The following tests passed:
	iree/tests/e2e/matmul/e2e_matmul_mmt4d_i8_large_llvm-cpu_local-task_dotprod

100% tests passed, 0 tests failed out of 1

Label Time Summary:
+dotprod                 =   0.11 sec*proc (1 test)
driver=local-task        =   0.11 sec*proc (1 test)
iree/tests/e2e/matmul    =   0.11 sec*proc (1 test)

Total Test time (real) =   0.23 sec
```

**Comment by bjacob** [comment]
Thank you very much @wangkuiyi !!

I'm sorry that you had to deal with our setup instructions not allowing to reproduce something that CI does.  If you can narrow this down to specifically the relevant difference, that would be great to fix (at least as a doc fix, expand the docs to explain optional things that enable building these tests).
